go/engine/unlock_test.go

```diff
@@ -77,3 +77,38 @@ func TestUnlockNoop(t *testing.T) {
 		t.Fatal("expected valid stream cache after unlock")
 	}
 }
+
+func TestUnlockWithPassphrase(t *testing.T) {
+	tc := SetupEngineTest(t, "template")
+	defer tc.Cleanup()
+
+	fu := CreateAndSignupFakeUser(tc, "login")
+
+	if !assertStreamCache(tc, true) {
+		t.Fatal("expected valid stream cache after sign up")
+	}
+
+	ctx := &Context{
+		LogUI:   tc.G.UI.GetLogUI(),
+		LoginUI: &libkb.TestLoginUI{},
+		// This SecretUI should not get used.
+		SecretUI: &libkb.TestSecretUI{Passphrase: "WRONG PASSPHRASE"},
```

Comment:
you could change engine/unlock.go RequiredUIs to return nil if e.passphrase != "".